### ISSUE TITLE
campaign_id parameter causes an error if the associated campaign does not exist

### ISSUE THREAD

**Issue by MarcMichalsky** [systopia/de.systopia.twingle#43]
If a `campaign_id` is passed as parameter to the TwingleDonation.submit API and there is no associated campaign for it, this causes an error:

```json
{
    "is_error": 1,
    "error_message": "'2' is not a valid option for field campaign_id"
}
``` 

I already have a fix for this – just needed the issue id for my branch. ;)

**Comment by jensschuppe** [comment]
The question is whether this should silently fail, fall back to the default campaign in the profile (if any), or propagate the error to the caller (as currently done).

Since the parameter is user input, I'd suggest to fall back to the default campaign, but add a description to that field in the configuration form.

**Comment by MarcMichalsky** [comment]
> Since the parameter is user input, I'd suggest to fall back to the default campaign, but add a description to that field in the configuration form.

That's what my PR #45 does. If there's no campaign with that id, the parameter gets unset and the default campaign is getting used instead. I don't think, a description is needed in the config form because the contribution does not get lost in no case.

**Comment by jensschuppe** [comment]
Released with `1.3-beta1`.
